ros/api/v1/hosts.py

```diff
@@ -376,57 +380,142 @@ def get(self, host_id):
 
 class ExecutiveReportAPI(Resource):
 
+    count_and_percentage = {
+        "count": fields.Integer,
+        "percentage": fields.Float
+    }
+    systems_per_state = {
+        "under_pressure": fields.Nested(count_and_percentage),
+        "undersized": fields.Nested(count_and_percentage),
+        "oversized": fields.Nested(count_and_percentage),
+        "waiting_for_data": fields.Nested(count_and_percentage),
+        "idling": fields.Nested(count_and_percentage)
+    }
+    condtions = {
+        "cpu": fields.Raw,
+        "io": fields.Raw,
+        "memory": fields.Raw
+    }
+    meta = {
+        "total_count": fields.Integer,
+        "non_optimized_count": fields.Integer,
+        "conditions_count": fields.Integer,
+    }
+    report_fields = {
+        "systems_per_state": fields.Nested(systems_per_state),
+        "conditions": fields.Nested(condtions),
+        "meta": fields.Nested(meta)
+    }
+
+    @marshal_with(report_fields)
     def get(self):
+        account_number = identity(request)['identity']['account_number']
+        system_queryset = system_ids_by_account(account_number, fetch_records=True)
+
+        # System counts
+        all_systems_count = system_queryset.count()
+        under_pressure_systems = count_per_state(system_queryset, {'state': "Under pressure"})
+        undersized_systems = count_per_state(system_queryset, {'state': "Undersized"})
+        oversized_systems = count_per_state(system_queryset, {'state': "Oversized"})
+        awaiting_data_systems = count_per_state(system_queryset, {'state': "Waiting for data"})
+        idling_systems = count_per_state(system_queryset, {'state': "Idling"})
+
+        non_optimized = system_queryset.filter(
+            or_(
+                System.cpu_state is not None,
+                System.io_state is not None,
+                System.memory_state is not None,
+            ),
+            and_(
+                System.state != 'Optimized',
+                System.state != 'Waiting for data'
+            )
+        )
+        non_optimized_count = non_optimized.count()
+
+        # cpu counts
+        total_cpu_issues = sum([
```

Comment:
What does the term `issues` signify ?

Reply:
Problematic sub-states of cpu component per system